### PR TITLE
Add PHPStan, lint and composer audit to CI

### PR DESCRIPTION
**Pull request by sf-cg**
In the github actions CI we've added:

* Lint Check
*  PHPStan check (and added the dependency for it to composer)
*  Composer audit check

On top of this, I've run PHPStan locally and cleaned up the issues it has initially shown.

### CODE REVIEW

**Review by girishl-cg — COMMENTED** [review]
@sf-cg The audit, lint, phpstan and tests has to be separated out  I think so you can see them individually.

**Comment by sf-cg** [comment]
@girishl-cg  I've updated so that Lint, Composer Scans, Static Analysis and Tests run separately.

File: .github/workflows/composer.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    strategy:
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@master
+
+      - name: setup php
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: 8.2
```
Comment (girishl-cg) [inline .github/workflows/composer.yml]: This should be a ${{ matrix.php-version }}

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    strategy:
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@master
+
+      - name: setup php
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: 8.2
```
Comment (girishl-cg) [inline .github/workflows/lint.yml]: same should be a ${{ matrix.php-version }}

File: .github/workflows/static.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    strategy:
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@master
+
+      - name: setup php
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: 8.2
```
Comment (girishl-cg) [inline .github/workflows/static.yml]: shd be a ${{ matrix.php-version }}

**Review by girishl-cg — CHANGES_REQUESTED** [review]

**Review by girishl-cg — APPROVED** [review]

**Pull request merged**